bin/generate_event.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Generates a sample S3 event for testing pigeon invocation.
+#
+# Usage: ./bin/generate_event.py --key=KEY
+
+import argparse
+import json
+import sys
+
+
+def make_event(key, event_name='ObjectCreated:Put', bucket='us-west-2'):
+    """Generates an S3 event
+
+    .. Note::
+
+       This only generates enough event boilerplate for pigeon.
+
+    """
+    return {
+        'Records': [
+            {
+                'eventVersion': '2.0',
+                'eventSource': 'aws:s3',
+                'eventName': event_name,
+                's3': {
+                    's3SchemaVersion': '1.0',
+                    'object': {
+                        'key': key
+                    },
+                    'bucket': {
+                        'arn': 'arn:aws:s3:::' + bucket,
+                        'name': bucket,
+                        'ownerIdentity': {
+                            'principalId': 'pigeonrules'
+                        }
+                    }
+                }
+            }
+        ]
+    }
```

Comment:
The full event is much bigger, but I skipped most of it since it's not interesting to `pigeon`.